mongodbatlas/provider.go

```diff
@@ -219,7 +215,6 @@ func getResourcesMap() map[string]*schema.Resource {
 		"mongodbatlas_privatelink_endpoint":                                        resourceMongoDBAtlasPrivateLinkEndpoint(),
 		"mongodbatlas_privatelink_endpoint_serverless":                             resourceMongoDBAtlasPrivateLinkEndpointServerless(),
 		"mongodbatlas_privatelink_endpoint_service":                                resourceMongoDBAtlasPrivateEndpointServiceLink(),
-		"mongodbatlas_privatelink_endpoint_service_adl":                            resourceMongoDBAtlasPrivateLinkEndpointServiceADL(),
```

Comment:
Something to consider: Even though we have the deprecation message specifying removal in 1.12.0 version, I am concerned in including the removal of complete resources within a minor version `1.12.2`. As a user updating to a minor version, I would expect a guarantee of no breaking changes.
If we have no urgency with these removals, we could schedule for `1.13.0` as a safer bet.

Reply:
I think that's something we discussed already, I believe in the risk you are mentioning but I thought we justified it by saying that in any case this was declared to be removed in 1.12.0, so no matter what those customers would have been already impacted and they should have already updated it.

of course these are cases where a little bit of telemetry (e.g. how many are using this resource?) would have been great.